modules/fluid_properties/src/fluidproperties/IdealRealGasMixtureFluidProperties.C

```diff
@@ -14,8 +14,224 @@
+    dy_dx.resize(_n_secondary_vapors);                                                             \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+      dy_dx[i] = dy_dx_Tv[i] + dy_dv_T * dv_dx[i];                                                 \
+  }
+
+/**
+ * Defines the value and derivative methods from (T,v) for a mass-specific property y
+ */
+#define define_mass_specific_prop_from_T_v(prop)                                                   \
+  Real IdealRealGasMixtureFluidProperties::prop##_from_T_v(                                        \
+      Real T, Real v, const std::vector<Real> & x) const                                           \
+  {                                                                                                \
+    const Real x_primary = primaryMassFraction(x);                                                 \
+    Real y = x_primary * _fp_primary->prop##_from_T_v(T, v / x_primary);                           \
+                                                                                                   \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+      y += x[i] * _fp_secondary[i]->prop##_from_T_v(T, v / x[i]);                                  \
+                                                                                                   \
+    return y;                                                                                      \
+  }                                                                                                \
+                                                                                                   \
+  void IdealRealGasMixtureFluidProperties::prop##_from_T_v(Real T,                                 \
+                                                           Real v,                                 \
+                                                           const std::vector<Real> & x,            \
+                                                           Real & y,                               \
+                                                           Real & dy_dT,                           \
+                                                           Real & dy_dv,                           \
+                                                           std::vector<Real> & dy_dx) const        \
+  {                                                                                                \
+    const Real x_primary = primaryMassFraction(x);                                                 \
+                                                                                                   \
+    Real y_primary, dy_dT_primary, dy_dv_primary;                                                  \
+    _fp_primary->prop##_from_T_v(T, v / x_primary, y_primary, dy_dT_primary, dy_dv_primary);       \
```

Comment:
I dont have good insights on this. If x is tiny, v/x is huge which means very low pressure. wouldnt we bring fp_primary out of bounds easily that way?

Reply:
This is something that should be guarded, since `x = 0` is a valid input. I have not addressed this in my changes, only refactored the existing formulation.

Reply:
I'll put an assert that `x != 0`.